dali/operators/reader/numpy_reader_op.cc

```diff
@@ -144,6 +232,25 @@ bool NumpyReader::SetupImpl(std::vector<OutputDesc> &output_desc,
       for (int d = 0; d < ndim; d++)
         sample_sh[d] = file_sh[d];
     }
+
+    if (need_slice) {
+      // Calculate the cropping window, based on the final layout (user provides axes in that
+      // layout)
+      auto tmp_roi = slice_attr_.GetCropWindowGenerator(i)(sh.tensor_shape(i), {});
+      sh.set_tensor_shape(i, tmp_roi.shape);  // set the final shape
+
+      // Reverse the cropping window arguments if needed, as we do slice before transpose to save
```

Comment:
Now you do slice and transpose in one go, I think you should update the comment accordingly.